File: k9mail/src/main/java/com/fsck/k9/fragment/MessageViewHolder.java
package com.fsck.k9.fragment;


import android.view.View;
import android.widget.CheckBox;
import android.widget.TextView;

import com.fsck.k9.R;
import com.fsck.k9.ui.ContactBadge;


public class MessageViewHolder implements View.OnClickListener {
    private final MessageListFragment fragment;
    public TextView subject;
    public TextView preview;
    public TextView from;
    public TextView time;
    public TextView date;
    public View chip;
    public TextView threadCount;
    public CheckBox flagged;
    public CheckBox selected;
    public int position = -1;
    public ContactBadge contactBadge;

    public MessageViewHolder(MessageListFragment fragment) {
        this.fragment = fragment;
    }

    @Override
    public void onClick(View view) {
        if (position != -1) {
            switch (view.getId()) {
                case R.id.selected_checkbox:
                    fragment.toggleMessageSelectWithAdapterPosition(position);
                    break;
                case R.id.flagged_bottom_right:
                case R.id.flagged_center_right:
                    fragment.toggleMessageFlagWithAdapterPosition(position);
                    break;
            }
        }
    }
}

File: app/ui/legacy/src/main/java/com/fsck/k9/fragment/MessageViewHolder.kt
package com.fsck.k9.fragment

import android.view.View
import android.widget.CheckBox
import android.widget.ImageView
import android.widget.TextView
import com.fsck.k9.ui.R

class MessageViewHolder(view: View) {
    var position = -1

    val selected: View = view.findViewById(R.id.selected)
    val contactPicture: ImageView = view.findViewById(R.id.contact_picture)
    val subject: TextView = view.findViewById(R.id.subject)
    val preview: TextView = view.findViewById(R.id.preview)
    val date: TextView = view.findViewById(R.id.date)
    val chip: ImageView = view.findViewById(R.id.account_color_chip)
    val threadCount: TextView = view.findViewById(R.id.thread_count)
    val flagged: CheckBox = view.findViewById(R.id.star)
    val attachment: ImageView = view.findViewById(R.id.attachment)
    val status: ImageView = view.findViewById(R.id.status)
}
